src/app/modules/group/group-detail/group-detail.component.ts

```diff
@@ -1,70 +1,166 @@
 import { Component, Inject, OnInit } from '@angular/core';
 import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
 import { MatDialog } from '@angular/material/dialog';
+import { VolunteerGroupService } from 'src/app/core/http/volunteer-group.service';
+import { NotificationService } from 'src/app/shared/components/notification/notification.service';
 import { DeleteGroupComponent } from './delete-group/delete-group.component';
+import { SearchMemberComponent } from './search-member/search-member.component';
 import { UpdateAddressComponent } from './update-address/update-address.component';
 import { UpdateNameComponent } from './update-name/update-name.component';
 import { UpdatePhoneComponent } from './update-phone/update-phone.component';
 import { UpdateSupportComponent } from './update-support/update-support.component';
+import { S3Service } from "../../../core/services/s3.service";
 
 @Component({
   selector: 'app-group-detail',
   templateUrl: './group-detail.component.html',
   styleUrls: ['./group-detail.component.scss'],
 })
 export class GroupDetailComponent implements OnInit {
+
+  currentInput:any;
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public group: IVolunteerGroup,
+    public dialogRef: MatDialogRef<GroupDetailComponent>,
+    public dialog: MatDialog,
+    private GroupService: VolunteerGroupService,
+    private notification: NotificationService,
+    private s3Service: S3Service
+  ) {}
+
+  ngOnInit(): void {}
+
   onClose() {
     this.dialogRef.close();
   }
 
   openUpdateName(cur_name: any, id: any) {
     this.dialog.open(UpdateNameComponent, {
       panelClass: 'nameType',
+      disableClose: true,
       data: { cur_name: cur_name, id: id },
+    }).afterClosed().subscribe((result: any)=>{
+      if(result){
+        this.group.name = result.data.name;
```

Comment:
mấy chỗ update này em cẩn thận, phải update lại detail nhưng khi thoát cái detail ra ngoài list cũng phải update lại item trong cái list. khi đóng cái bottom sheet thì gửi nguyên cái group object ra ngoài. sau đó dùng hàm map để update lại cái list: `this.groups = groups.map(element => group.id === element.id ? group: element);`

Reply:
với lại chỗ này nếu mà backend trả về nguyên cái group thì chỉ cần update như sau:
`this.group = result.data`